pkg/providers/snow/objects.go

```diff
@@ -0,0 +1,126 @@
+package snow
+
+import (
+	"context"
+
+	"k8s.io/apimachinery/pkg/api/equality"
+	"k8s.io/apimachinery/pkg/runtime"
+	clusterv1 "sigs.k8s.io/cluster-api/api/v1beta1"
+	"sigs.k8s.io/cluster-api/bootstrap/kubeadm/api/v1beta1"
+
+	"github.com/aws/eks-anywhere/pkg/cluster"
+	"github.com/aws/eks-anywhere/pkg/providers/common"
+	snowv1 "github.com/aws/eks-anywhere/pkg/providers/snow/api/v1beta1"
+)
+
+func ControlPlaneObjectsForCreate(clusterSpec *cluster.Spec) ([]runtime.Object, error) {
+	snowCluster := SnowCluster(clusterSpec)
+	controlPlaneMachineTemplate := SnowMachineTemplate(clusterSpec.SnowMachineConfigs[clusterSpec.Cluster.Spec.ControlPlaneConfiguration.MachineGroupRef.Name])
+	kubeadmControlPlane, err := KubeadmControlPlane(clusterSpec, controlPlaneMachineTemplate)
+	if err != nil {
+		return nil, err
+	}
+	capiCluster := CAPICluster(clusterSpec, snowCluster, kubeadmControlPlane)
+
+	return []runtime.Object{capiCluster, snowCluster, kubeadmControlPlane, controlPlaneMachineTemplate}, nil
+}
+
+func WorkersObjectsForCreate(clusterSpec *cluster.Spec) ([]runtime.Object, error) {
+	kubeadmConfigTemplates, err := KubeadmConfigTemplates(clusterSpec)
+	if err != nil {
+		return nil, err
+	}
+	workerMachineTemplates := SnowMachineTemplates(clusterSpec)
+	machineDeployments := MachineDeployments(clusterSpec, kubeadmConfigTemplates, workerMachineTemplates)
+
+	return concatWorkersObjects(machineDeployments, kubeadmConfigTemplates, workerMachineTemplates), nil
+}
+
+func ControlPlaneObjectsForUpgrade(ctx context.Context, oldSpec, newSpec *cluster.Spec, kubeClient KubeClient) ([]runtime.Object, error) {
```

Comment:
What do we need the `oldSpec` for? I thought comparing the output avoided needed the old Spec
Ideally we should avoid that, because it's no going to available in the controller

Reply:
we need it to fetch the old cp/md in order to get old machinetemplates for comparision